### ISSUE TITLE
Implement ScaleEmbedding nn.Module

### ISSUE THREAD

**Issue by timlawrenz** [timlawrenz/DINO-X#1]
## Summary
Implement a lightweight `ScaleEmbedding` MLP that maps 3 physical-space floats to the ViT embedding dimension.

## Design

```python
class ScaleEmbedding(nn.Module):
    """Continuous scale embedding for physical-space awareness."""
    # Input: (pixel_spacing_x, pixel_spacing_y, slice_thickness) — 3 floats
    # Architecture: Linear(3, embed_dim//4) → GELU → Linear(embed_dim//4, embed_dim) → LayerNorm
    # Output: (batch, embed_dim) — broadcast-added to every patch embedding
```

## Key Decisions
- **Continuous, not categorical** — inspired by FlexiViT/VariViT. Model generalizes to unseen spacings (trained on 0.5mm and 1.5mm, works at 0.8mm)
- **Added to patch embeddings, not concatenated** — preserves sequence length, compatible with positional embeddings
- **Small MLP (one hidden layer)** — learns non-linear spacing relationships (e.g., anisotropy)
- **LayerNorm before addition** — prevents scale embedding from dominating patch features early
- **Default spacing = (1.0, 1.0, 1.0)** — backward-compatible with existing checkpoints

## Acceptance Criteria
- [ ] `ScaleEmbedding(embed_dim=384)` produces correct output shape `(B, 384)`
- [ ] Gradients flow correctly through the module
- [ ] Default spacing (1,1,1) produces near-zero contribution after LayerNorm init
- [ ] Unit tests for shape, gradient flow, device placement

**Comment by timlawrenz** [comment]
✅ Implemented in `scripts/phase5_big_run.py` as `ScaleEmbedding` class.

MLP architecture: `Linear(3→dim//4) → GELU → Linear(dim//4→dim) → LayerNorm`, output zero-initialized. For ViT-Small (dim=384): hidden=96, 37K params (0.15% of backbone). Accepts continuous (pixel_spacing_x, pixel_spacing_y, slice_thickness) and adds result to patch embeddings.

Commit: e50f3b2
